File: spec/python/test_instance_std_array.py
import unittest

from instance_std_array import InstanceStdArray

class TestInstanceStdArray(unittest.TestCase):
    def test_instance_std_array(self):
        with InstanceStdArray.from_file("src/instance_std_array.bin") as r:

            self.assertEqual(r.ofs, 0x10)
            self.assertEqual(r.qty_entries, 3)
            self.assertEqual(r.entry_size, 4)

            self.assertEqual(r.entries, [
                b"\x11\x11\x11\x11",
                b"\x22\x22\x22\x22",
                b"\x33\x33\x33\x33",
            ])

File: spec/lua/test_instance_std_array.lua
local luaunit = require("luaunit")

require("instance_std_array")

TestInstanceStdArray = {}

function TestInstanceStdArray:test_instance_std_array()
    local r = InstanceStdArray:from_file("src/instance_std_array.bin")

    luaunit.assertEquals(r.ofs, 0x10)
    luaunit.assertEquals(r.qty_entries, 3)
    luaunit.assertEquals(r.entry_size, 4)

    luaunit.assertEquals(r.entries, {
        "\x11\x11\x11\x11",
        "\x22\x22\x22\x22",
        "\x33\x33\x33\x33",
    })
end
